src/DefaultBuilder/src/WebHostEnvironment.cs

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+using System.Reflection;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Hosting.StaticWebAssets;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.FileProviders;
+using Microsoft.Extensions.Hosting;
+
+namespace Microsoft.AspNetCore.Builder
+{
+    internal class WebHostEnvironment : IWebHostEnvironment
+    {
+        private static readonly NullFileProvider NullFileProvider = new();
+
+        public WebHostEnvironment(Assembly? callingAssembly)
+        {
+            ContentRootPath = Directory.GetCurrentDirectory();
+
+            ApplicationName = (callingAssembly ?? Assembly.GetEntryAssembly())?.GetName()?.Name ?? "NotFound";
+            EnvironmentName = Environments.Production;
```

Comment:
Is this supposed to always be Production?

Reply:
I guess that's the wrong question.

Should host config override webhost config? If you run the DefaultBuilder.SampleApp you can see it has ASPNETCORE_ENVIRONMENT set to Development but gets set to Production at https://github.com/dotnet/aspnetcore/pull/31825/files#diff-f0e6f715f99a336de90bdbeef327cb71b7ee811d00d7c1722049557adaa009c5R168

Reply:
Initially, that's the default. It will be [overridden by environment variables or command line arguments by ConfigureDefaults()](https://github.com/dotnet/runtime/blob/main/src/libraries/Microsoft.Extensions.Hosting/src/HostingHostBuilderExtensions.cs#L164-L171). The flow here is pretty wild.

After the WebHostEnvironment is constructed in the WebApplicationBuilder constructor, we pass it to the ConfigureHostBuilder constructor that [calls ConfigureDefaults() on itself](https://github.com/dotnet/aspnetcore/blob/eb4ef57600c6e345fb911e341c8458ffd05c41ff/src/DefaultBuilder/src/ConfigureHostBuilder.cs#L37). ConfigureDefaults() calls ConfigureHostBuilder.ConfigureHostConfiguration which [sets the EnvironmentName as appropriate](https://github.com/dotnet/aspnetcore/blob/eb4ef57600c6e345fb911e341c8458ffd05c41ff/src/DefaultBuilder/src/ConfigureHostBuilder.cs#L69).

I'm thinking of ways to simplify the flow while keeping the shared logic in ConfigureDefaults().

Reply:
I didn't see your first comment when I first replied. `DOTNET_` environment variables worked because we were bootstrapping  `ConfigureDefaults` but `ASPNETCORE_` environment variables did not work because we were not bootstrapping `ConfigureWebHostDefaults`. This is fixed by https://github.com/dotnet/aspnetcore/pull/31825/commits/b2ab79d0e930a1f03955f233aea901d911775894.